force-app/main/default/lwc/productTileList/__tests__/productTileList.test.js

```diff
@@ -0,0 +1,215 @@
+import { createElement } from 'lwc';
+import ProductTileList from 'c/productTileList';
+import { fireEvent } from 'c/pubsub';
+import {
+    registerTestWireAdapter,
+    registerLdsTestWireAdapter,
+} from '@salesforce/wire-service-jest-util';
+import getProducts from '@salesforce/apex/ProductController.getProducts';
+import { CurrentPageReference } from 'lightning/navigation';
+
+// Mock out the pubsub lib and use these mocks to verify how functions were called
+jest.mock('c/pubsub', () => {
+    return {
+        fireEvent: jest.fn(),
+        registerListener: jest.fn(),
+        unregisterAllListeners: jest.fn(),
+    };
+});
+
+// TODO(tbliss): the default resolution for @salesforce/apex/foo.bar is a function that returns a resolved Promise.
+//               that errors when it's used as an @wire id. The import needs to have object identity in the component
+//               and the test.
+jest.mock(
+    '@salesforce/apex/ProductController.getProducts',
+    () => {
+        return { default: jest.fn() };
+    },
+    { virtual: true },
+);
+
+// realistic data with multiple records
+const mockGetProducts = require('./data/getProducts.json');
+// an empty list of records to verify the component does something reasonable
+// when there is no data ato display
+const mockGetProductsNoRecords = require('./data/getProductsNoRecords.json');
+
+// Register as an LDS wire adapter. Some tests verify the provisioned values trigger desired behavior.
+const getProductsAdapter = registerLdsTestWireAdapter(getProducts);
+
+// Register as a standard wire adapter because the component under test requires this adapter.
+// We don't exercise this wire adapter in the tests.
+registerTestWireAdapter(CurrentPageReference);
+
+describe('c-product-tile-list', () => {
+    afterEach(() => {
+        // The jsdom instance is shared across test cases in a single file so reset the DOM
+        while (document.body.firstChild) {
+            document.body.removeChild(document.body.firstChild);
+        }
+    });
+
+    describe('with records', () => {
+        it('renders paginator with correct item counts', () => {
+            const element = createElement('c-product-tile-list', {
+                is: ProductTileList,
+            });
+            document.body.appendChild(element);
+            getProductsAdapter.emit(mockGetProducts);
+
+            // Return a promise to wait for any asynchronous DOM updates.
+            return Promise.resolve().then(() => {
+                const paginator = element.shadowRoot.querySelector(
+                    'c-paginator',
+                );
+                expect(paginator).not.toBeNull();
+                // we know from the mock data (./data/getProducts.json) we
+                // should have 12 total items and a page size of 9
```

Comment:
Why hard code this instead of using the mock data? Eg `mockGetProducts.pageSize` and `mockGetProducts.totalItemCount`